### DISCUSSION TITLE
Use Cloudflare's worker to handle the requests

### DISCUSSION THREAD

**Discussion by itmanpro** [mymmrac/telego#132] [General]
Hello!
Thank you for all your efforts and grate work!
I'm using a project and in that project your project included, so I need your help about something.
Telegram is restricted in my country and I need to handle all request by some proxy or a tool like Cloudflare's worker
could you please help me about that?

I tried a GitHub project and made a worker on Cloudflare:

https://github.com/manzoorwanijk/telegram-bot-api-worker

and replaced all the Telegram links in your project with the links of that worker, but it didn't work, I'm getting an error:

ERROR Execution error getUpdates: request call: fasthttp do request: tls: failed to verify certificate: x509: certificate is valid for *.mywebsite.com, mywebsite.com, not api.telegram.org

**Comment by mymmrac** [comment]
Hi, interesting use case, I think that the issues is in the request you do on fetch - you are using the original request and not creating fully new one. That may result in some inconsistency, it seems like when you returned the result from Cloudflare it contained data related to `api.telegram.org` and not for your site.

Try instead of `new Request(url.toString(), request);` use something like this:
```js
new Request(url.toString(), {
    method: request.method,
    headers: request.headers,
    // Add other request options if necessary
  });
``` 

You may need to add some other request options or alter the request headers (I am not sure actually), but it also may work as is/

If even this doesn't work, try to create a new response and not return the same from fetch, too.

**Comment by itmanpro** [comment]
Hello again, I found that the message I was getting from my server about the certificate problem, actually was occurring because of my settings in the hosts file some days ago, I made a change to the hosts file for testing a solution but I forget to undo it, so the error message has been changed since last night, now I'm getting another kind of error message on my server:

ERROR Getting updates: telego: getUpdates(): internal execution: request call: fasthttp do request: dial tcp4 "The government's firewall IP":443: connect: connection refused

**Reply by mymmrac** [reply]
I think this error is related to your settings too, maybe you have to configure firewall to allow requests to port 443 or something like that

**Reply by itmanpro** [reply]
The 443 port along with some other ports, are the default https support ports of Cloudflare, there is no problem with Cloudflare, also on both my servers, firewall is down.
The only thing here is just, how can I change your code to use Cloudflare worker address instead of Telegram's original server addresses

**Reply by mymmrac** [reply]
This is easy, just add `telego.WithAPIServer("mywebsite.com")` option when creating the bot, here is example: https://github.com/mymmrac/telego/blob/53382e52bfecd3aba726eb1343ab56c392cffd77/examples/configuration/main.go#L22

**Reply by itmanpro** [reply]
That's it? just put my API address over there and finished?!
That's so easy man, Thank you very much!
I thought I need to replace every Telegram links on the Entire project!
Thanks again.

**Reply by itmanpro** [reply]
1) telego.WithAPIServer(https://somerandomname.workers.dev)

2) telego.WithAPIServer("https://somerandomname.workers.dev")

3) telego.WithAPIServer("somerandomname.workers.dev")

4) telego.WithAPIServer(somerandomname.workers.dev)

which one?

**Reply by mymmrac** [reply]
Second
